### PR TITLE
[diffusion]: MOVA torch.compile opt

### PR DESCRIPTION
**Pull request by triple-mu**
## Motivation

The previous implementation manually wrapped `module.forward` with `torch.compile`,
which is less idiomatic and may introduce subtle issues when interacting with
PyTorch internals.

## Modifications

- Refactor `compile_module_with_torch_compile` into `_maybe_enable_torch_compile`
- Switch to the official `nn.Module.compile()` API instead of patching `forward`
- Add explicit `isinstance(module, nn.Module)` check
- Remove unnecessary return value (in-place behavior)
- Improve type annotations by importing `torch.nn as nn`
- Minor cleanup and logging improvements

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/mova.py [hunk 1]
```diff
@@ -17,6 +17,7 @@
 from collections.abc import Iterable
 
 import torch
+import torch.nn as nn
```
Comment (ping1jing2) [inline python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/mova.py:20]: Could you please help move all type annotations into TYPE_CHECKING block like code below :
```python
if TYPE_CHECKING:
    from sglang.multimodal_gen.runtime.pipelines_core.schedule_batch import  Req
    from sglang.multimodal_gen.runtime.server_args import ServerArgs
    xxxxx

```

**Review by mickqian — APPROVED** [review]

File: python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/mova.py (hunk 1 above)
Comment (triple-mu) [reply python/sglang/multimodal_gen/runtime/pipelines_core/stages/model_specific_stages/mova.py:20]: Ok, got it
